x-pack/plugins/security_solution/public/common/components/user_profiles/use_suggest_users.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+
+import type { UserProfileWithAvatar } from '@kbn/user-profile-components';
+
+import { suggestUsers } from './api';
+import { USER_PROFILES_FAILURE } from './translations';
+import { useAppToasts } from '../../hooks/use_app_toasts';
+
+export interface SuggestUserProfilesArgs {
+  searchTerm: string;
+}
+
+export const bulkGetUserProfiles = async ({
+  searchTerm,
+}: {
+  searchTerm: string;
+}): Promise<UserProfileWithAvatar[]> => {
+  return suggestUsers({ searchTerm });
+};
+
+export const useSuggestUsers = ({ searchTerm }: { searchTerm: string }) => {
```

Comment:
Those three are using different endpoints as you mentioned and also do different things. Unfortunately, we cannot reuse one for all of the cases. We could potentially re-use `use_bulk_get_user_profiles` for fetching current user if know it's UID, but I feel like it will be unnecessary enhancement.

`use_suggest_users` works in a way that it provides the list of available users (defaults to 10 users) in the project and allows to search for more.

`use_bulk_get_user_profiles` allows us to fetch the list of known users (known UIDs) which is beneficial in case where we have all UIDs, like in case with alerts table where we display users avatars.